### PR TITLE
add custom embed model selection, improve readme, update examples

### PR DESCRIPTION
**Pull request by SeeknnDestroy**
Refined embedding model selection in `AutoQueryEngine` to include Hugging Face pre-trained models and applied minor codebase fixes. It takes `embed_model` parameter for embedding model selection.

### CODE REVIEW

**Comment by fcakyon** [comment]
Could be great if we could update the quickstart demo to use `read_sitemap_as_documents` instead of the current usage. @SeeknnDestroy

File: autollm/auto/query_engine.py [hunk 1]
```diff
@@ -167,6 +171,7 @@ def from_config(
             system_prompt=config.get('system_prompt'),
             query_wrapper_prompt=config.get('query_wrapper_prompt'),
             enable_cost_calculator=config.get('enable_cost_calculator'),
+            embed_model=config.get('embed_model'),
```
Comment (fcakyon) [inline autollm/auto/query_engine.py:174]:
```suggestion
            embed_model=config.get('embed_model', 'default'),
```

File: autollm/auto/query_engine.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def create_query_engine(
         system_prompt (str): The system prompt to use for the query engine.
         query_wrapper_prompt (str): The query wrapper prompt to use for the query engine.
         enable_cost_calculator (bool): Flag to enable cost calculator logging.
-        embed_model (BaseEmbedding): The embedding model to use for the query engine. Defaults to OpenAIEmbedding.
+        embed_model (Union[str, EmbedType]): The embedding model to use for generating embeddings. Defaults to OpenAI.
```
Comment (fcakyon) [inline autollm/auto/query_engine.py:31]: we should add example usage for custom hf models as `local:intfloat/multilingual-e5-large`

File: autollm/auto/query_engine.py [hunk 3]
```diff
@@ -122,6 +124,7 @@ def from_parameters(
             system_prompt (str): The system prompt to use for the query engine.
             query_wrapper_prompt (str): The query wrapper prompt to use for the query engine.
             enable_cost_calculator (bool): Flag to enable cost calculator logging.
+            embed_model (Union[str, EmbedType]): The embedding model to use for generating embeddings. Defaults to OpenAI.
```
Comment (fcakyon) [inline autollm/auto/query_engine.py:127]: we should add example for custom hf model usage as `local:intfloat/multilingual-e5-large`

File: examples/configs/config.example.yaml [hunk 4]
```diff
@@ -1,10 +1,11 @@
 # config.example.yaml
-version: '1.0'  # Version of this configuration file
+version: '2.0'  # Version of this configuration file
 tasks:
   - name: "summarize"
     system_prompt: "You are an expert ai assistant specialized in summarization."  # System prompt for this task
     vector_store_params:
       vector_store_type: "SimpleVectorStore"
+    embed_model: "default"  # ["default", "local"]
```
Comment (fcakyon) [inline examples/configs/config.example.yaml:8]:
```suggestion
    embed_model: "default"  # ["default", "local:intfloat/multilingual-e5-large"]
```

File: README.md [hunk 5]
```diff
@@ -66,9 +66,9 @@ ______________________________________________________________________
 ### create a query engine in seconds
 
 ```python
->>> from autollm import AutoQueryEngine, read_files_as_documents
+>>> from autollm import AutoQueryEngine, read_sitemap_as_documents
 
->>> documents = read_files_as_documents(input_dir="examples/data")
```
Comment (fcakyon) [inline README.md:71]: both should stay, lets not remove `read_files_as_documents`

File: autollm/auto/vector_store_index.py [hunk 6]
```diff
@@ -64,17 +64,20 @@ def from_defaults(
         # Initialize vector store index from documents
         else:
             if vector_store_type == "LanceDBVectorStore" and "uri" not in kwargs:
-                kwargs["uri"] = "/tmp/lancedb"
+                kwargs["uri"] = "./.lancedb"
```
Comment (fcakyon) [inline autollm/auto/vector_store_index.py:67]: bunu readme ve notebookta da guncellemek lazim

Comment (SeeknnDestroy) [reply autollm/auto/vector_store_index.py]: bunu readme veya notebookta kullanmıyoruz ki @fcakyon abi?

Comment (fcakyon) [reply autollm/auto/vector_store_index.py]: readmede kullandigimiz yerlerden bir tanesi: https://github.com/safevideo/autollm/blob/12fb757f76ec274814eec208c25b3e1ed87e6c6e/README.md?plain=1#L96

Comment (fcakyon) [reply autollm/auto/vector_store_index.py]: readmede kullandigimiz yerlerden bir tanesi: https://github.com/safevideo/autollm/blob/12fb757f76ec274814eec208c25b3e1ed87e6c6e/README.md?plain=1#L96

Comment (fcakyon) [reply autollm/auto/vector_store_index.py]: bu da notebookta kullandigimiz yer: 
image

File: autollm/utils/document_reading.py [hunk 7]
```diff
@@ -112,3 +112,39 @@ def read_github_repo_as_documents(
         shutil.rmtree(temp_dir, onerror=on_rm_error)
 
     return documents
+
+
+def read_sitemap_as_documents(sitemap_url: str, filter: Optional[str] = None) -> List[Document]:
```
Comment (fcakyon) [inline autollm/utils/document_reading.py:117]: we can remove this reader

**Review by fcakyon — APPROVED** [review]

**Pull request merged**